solutions/gold/ac-swap.mdx

```diff
@@ -169,6 +169,90 @@ int main() {
 ```
 
 </CPPSection>
+
+<PySection>
+
+```py
+LETTERS = 3
+MAX_SWAPS = 465  # see hint above
+
+s = input()
+k = int(input())
+
+n = len(s)
+count = [0] * LETTERS
+for x in s:
+	if x == "K":
+		count[0] += 1
+	elif x == "E":
+		count[1] += 1
+	else:
+		count[2] += 1
+
+# stores the number of occurrences of a letter on a prefix
+pref = [[0] * LETTERS for _ in range(n + 1)]
+# stores the position of the ith letter
+pos = [[] for _ in range(LETTERS)]
+
+for i in range(n):
+	pref[i][0] = (pref[i - 1][0] if i - 1 >= 0 else 0) + (s[i] == "K")
+	pref[i][1] = (pref[i - 1][1] if i - 1 >= 0 else 0) + (s[i] == "E")
+	pref[i][2] = (pref[i - 1][2] if i - 1 >= 0 else 0) + (s[i] == "Y")
+
+	if s[i] == "K":
+		pos[0].append(i)
+	elif s[i] == "E":
+		pos[1].append(i)
+	else:
+		pos[2].append(i)
+
+dp = [
+	[[[0] * (count[2] + 1) for _ in range(count[1] + 1)] for _ in range(count[0] + 1)]
```

Comment:
this is a little hard to read. could you replace it with something a bit more intuitive?

Reply:
ok its not that bad but at least to me it does look slightly confusing

Reply:
^

Reply:
it's 4D, I don't know any other way to initialize it other than list comprehension. I can add a comment to it like:
# dp[cur][ks][es][ys] = number of ways to achieve the state with ks 'K's, es 'E's, ys 'Y's used, and cur swaps made